Errors shown in AlertManagerConfig Receive create/edit page

<!--------- For bugs and general issues --------->
**Setup**
- Rancher version: v2.11.2 (and before?)

**Describe the bug**
- `actions.js:642 Attempting to find a resource with no id monitoring.coreos.com.alertmanagerconfig`
  - nav to AlertManagerConfig Create page
    - create/edit component - all bugs hidden by https://github.com/rancher/dashboard/issues/14944 (mostly around actions and routing)
    - receiverConfig component - all bugs hidden by https://github.com/rancher/dashboard/issues/14944 (unable to edit existing receiver configs)
- `Error in fetch(): TypeError: Cannot read properties of undefined (reading 'availableActions')`
  - nav to AlertManagerConfig Create page  - all bugs hidden by https://github.com/rancher/dashboard/issues/14944 (mostly around actions and routing)
- `fetch.client.js:76 Error in fetch(): Error: Can't render the form because the AlertmanagerConfig schema, or it's definitions, is not loaded yet.`
  - nav to existing AlertManagerConfig page and click add receiver  - all bugs hidden by https://github.com/rancher/dashboard/issues/14944 (unable to edit existing receiver configs)


**To Reproduce**
- as above

**Result**
- as above

**Expected Result**
- no errors (and functionality validated once https://github.com/rancher/dashboard/issues/14944 is resolved)

**Screenshots**

<img width="1391" height="791" alt="Image" src="https://github.com/user-attachments/assets/1ef17a14-a000-40c8-8f0e-8e9438b16fed" />

--- a/shell/edit/monitoring.coreos.com.alertmanagerconfig/index.vue
+++ b/shell/edit/monitoring.coreos.com.alertmanagerconfig/index.vue
@@ -34,23 +34,24 @@ export default {
 
   async fetch() {
     const inStore = this.$store.getters['currentProduct'].inStore;
-    const alertmanagerConfigId = this.value.id;
-
-    const { receiverSchema, routeSchema } = await fetchAlertManagerConfigSpecs(this.$store);
+    const { receiverSchema } = await fetchAlertManagerConfigSpecs(this.$store);
 
     this.receiverSchema = receiverSchema;
-    this.routeSchema = routeSchema;
 
-    const alertmanagerConfigResource = await this.$store.dispatch(`${ inStore }/find`, { type: MONITORING.ALERTMANAGERCONFIG, id: alertmanagerConfigId });
+    // The edit page is (mis)used as the detail page. When it's in create world none of the below is valid (there's no existing resource)
+    if (!this.isCreate) {
+      const alertmanagerConfigResource = await this.$store.dispatch(`${ inStore }/find`, { type: MONITORING.ALERTMANAGERCONFIG, id: this.value.id });
 
-    this.alertmanagerConfigId = alertmanagerConfigId;
-    this.alertmanagerConfigResource = alertmanagerConfigResource;
-    this.alertmanagerConfigDetailRoute = alertmanagerConfigResource?._detailLocation;
+      this.alertmanagerConfigId = this.value.id;
+      this.alertmanagerConfigResource = alertmanagerConfigResource;
 
-    const alertmanagerConfigActions = alertmanagerConfigResource.availableActions;
-    const receiverActions = alertmanagerConfigResource.getReceiverActions(alertmanagerConfigActions);
+      this.alertmanagerConfigDetailRoute = alertmanagerConfigResource._detailLocation;
 
-    this.receiverActions = receiverActions;
+      const alertmanagerConfigActions = alertmanagerConfigResource.availableActions;
+      const receiverActions = alertmanagerConfigResource.getReceiverActions(alertmanagerConfigActions);
+
+      this.receiverActions = receiverActions;
+    }
   },
 
   data() {

--- a/shell/edit/monitoring.coreos.com.alertmanagerconfig/receiverConfig.vue
+++ b/shell/edit/monitoring.coreos.com.alertmanagerconfig/receiverConfig.vue
@@ -128,6 +128,16 @@ export default {
     const expectedFields = Object.keys(receiverSchema.resourceFields);
     const suffix = {};
 
+    // Values contains multiple receivers of two types
+    // 1. Known (expected, from schema) shown in their own tabs with explicity forms
+    // 2. unknown (not expected, missing in schema) shown as a yaml blob
+
+    // - expectedFields and suffixYaml are only used if the receiver is of type `custom`
+    // - expectedFields are the known types
+    // - suffixYaml are the unknown types
+    // - usages,
+    //   - for custom, we need to know only the unknown / yaml blog. so suffixYaml is created by extracting all known (expectedFields) from value
+    //   - on edit of custom we then combine the two again and save to value
     Object.keys(this.value).forEach((key) => {
       if (!expectedFields.includes(key)) {
         suffix[key] = this.value[key];
@@ -195,7 +205,7 @@ export default {
       return {
         duplicateName: () => {
           const receiversArray = this.alertmanagerConfigResource.spec.receivers;
-          const receiverNamesArray = receiversArray.map((R) => R.name);
+          const receiverNamesArray = receiversArray?.map((R) => R.name) || [];
           const receiversSet = new Set(receiverNamesArray);
 
           if (receiversArray.length !== receiversSet.size) {

--- a/shell/utils/alertmanagerconfig.js
+++ b/shell/utils/alertmanagerconfig.js
@@ -19,8 +19,8 @@ export const fetchAlertManagerConfigSpecs = async($store) => {
   await schema.fetchResourceFields();
 
   return {
-    receiverSchema: schema.schemaDefinitions?.[`${ schema.schemaDefinition.id }.spec.receivers`],
-    routeSchema:    schema.schemaDefinitions?.[`${ schema.schemaDefinition.id }.spec.route`],
+    receiverSchema: schema.schemaDefinitions?.[`${ schema.schemaDefinition.type }.spec.receivers`],
+    routeSchema:    schema.schemaDefinitions?.[`${ schema.schemaDefinition.type }.spec.route`],
   };
 };
 

